src/error/mod.rs

```diff
@@ -321,6 +351,36 @@ where
     }
 }
 
+/// Error type used when a token for multiple recipients (i.e., `CoseEncrypt`) is decrypted.
+///
+/// In that case, the recipients may be encrypted with a different cipher (`K`) than the
+/// actual content (`C`); hence, this error type differentiates between the two.
+#[derive(Debug)]
+pub enum MultipleCoseError<K, C>
+where
+    K: Display,
+    C: Display,
+{
+    /// Used when an error occurred in the Key Encryption Key's cipher.
+    KekError(K),
+
+    /// Used when an error occurred in the Content Encryption Key's cipher.
+    CekError(C),
+}
```

Comment:
Is there any reason why this enum needs multiple generic parameters? As only one of the generic parameters is used at a time and both have the same type restrictions, you could probably just use a single generic type for both error variants.

This might also allow for factoring out common components of `from_kek_error` and `from_cek_error`.

Reply:
Yes, the need for this trait arose for the method `decrypt_access_token_multiple<K,C>`.
In it, a cipher for the KEK and a cipher for the CEK is used — both of these ciphers have their own error type. By having these two type parameters in `MultipleCoseError`, I can pass on both of these error types to the caller. If it only had a single parameter, this wouldn't be possible.

Reply:
Ah, in that case I don't see any obvious way of simplifying `from_kek_error` and `from_cek_error`